k2/csrc/rand.h

```diff
@@ -0,0 +1,70 @@
+// k2/csrc/rand.h
+/**
+ * @copyright
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#ifndef K2_CSRC_RAND_H_
+#define K2_CSRC_RAND_H_
+
+#include "k2/csrc/array.h"
+#include "k2/csrc/context.h"
+
+namespace k2 {
+
+/* Get the current seed of the device associated with `context`.
+ *
+ * @param [in] context  It specifies the device whose seed is to be returned.
+ *                      It can be either a CPU context or a CUDA context.
+ *
+ * @return  Return the seed of the device associated with the given `context`.
+ *
+ * TODO(fangjun): we may not need it.
+ */
+uint64_t GetSeed(ContextPtr context);
+
+/* Set the seed of the device associated with the given `context`.
+ *
+ * @param [in] context  It specifies the device whose seed is to be set.
+ *                      It can be either a CPU context or a CUDA context.
+ *
+ * @param [in] seed     The target seed.
+ */
+void SetSeed(ContextPtr context, uint64_t seed);
+
+/* Fill the given array with random numbers from a uniform distribution on
+ * the interval (0.0, 1.0].
+ *
+ * 0 is exclusive and 1 is inclusive.
```

Comment:
Wondering why do we need this version, i.e. (0, 1], I was thinking we don't really care the boundary and the version `[0,1)` would be enough

Reply:
this is also the default implementation of `uniform_real_distribution`

Reply:
`curand` returns (0, 1]. To make the behavior consistent for CPU and CUDA, `(0, 1]` is used here.

Reply:
Will try to reverse (0, 1] to [0, 1) for the CUDA part.